### ISSUE TITLE
Make `instance` return an instancetype to avoid conditional casting

### ISSUE THREAD

**Issue by ldiqual** [Appboy/appboy-segment-ios#3]
How about making `instance` return an `instancetype` instead of a generic `id` here? https://github.com/Appboy/appboy-segment-ios/blob/master/Pod/Classes/SEGAppboyIntegrationFactory.h#L6

This way we could avoid casting to `SEGAppboyIntegrationFactory` when setting up the segment configuration.

```
if let appboyFactory = SEGAppboyIntegrationFactory.instance() as? SEGAppboyIntegrationFactory {
    segmentConfiguration.use(appboyFactory)
}
```

**Comment by Wenzhi** [comment]
Hi @ldiqual,

I think it's a good suggestion, and we will put it in the next release.

Thank you!
Wenzhi

**Comment by Wenzhi** [comment]
Hi @ldiqual,

We just released version 1.0.2, which has the change.

Thanks for helping us improving our code!
Wenzhi

**Comment by ldiqual** [comment]
@Wenzhi Thanks! Much appreciated.
